shared/chat/conversation/input-area/normal/shared.js

```diff
@@ -0,0 +1,49 @@
+// @flow
+import * as React from 'react'
+import {Meta} from '../../../../common-adapters'
+import {globalColors, platformStyles, styleSheetCreate} from '../../../../styles'
+import {messageExplodeDescriptions} from '../../../../constants/chat2'
+
+export const ExplodingMeta = ({explodingModeSeconds}: {explodingModeSeconds: number}) => {
+  let title = 'New'
+
+  if (explodingModeSeconds !== 0) {
+    const description = messageExplodeDescriptions.find(
```

Comment:
I don't think this should look for a match in `messageExplodeDescriptions`. If it's set to something different than one of the choices, it doesn't show anything. Instead, this could do a couple checks on whether `explodingModeSeconds` is > a day, an hour, etc. and create a ballpark meta based on that. There are some similar functions in #12024 for displaying the countdown. Actually, `formatTimeDifference` could be a drop-in replacement for this, it just takes in milliseconds instead of seconds.

Reply:
That's a good idea. I dropped a comment on `formatTimeDifference` in your PR.

Reply:
I'll wait for #12024 to merged so I can use that helper.